Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -4241,13 +4241,48 @@ static bool isEmpty(PDFSelection *selection)
 
     RetainPtr selectionsByLine = [selection selectionsByLine];
     RetainPtr selectedLine = isStart ? [selectionsByLine firstObject] : [selectionsByLine lastObject];
-    FloatRect lineBounds = [selectedLine boundsForPage:page.get()];
-    if (lineBounds.isEmpty())
+    FloatRect boundsInRootView;
+
+    AffineTransform cumulativeTransform = [page transformForBox:kPDFDisplayBoxMediaBox];
+    bool appliedLineTransform = false;
+    [selectedLine enumerateRectsAndTransformsForPage:page.get() usingBlock:[&](CGRect rect, CGAffineTransform transform) {
+        if (std::exchange(appliedLineTransform, true))
```

Comment:
So we apply the transform for the first rect only?

Reply:
Ah, so in theory, there should only ever be a single rect (since this is enumerating rects along just the first line). I can add some assertions to help clarify (and enforce) this.

Reply:
Oh, right, that makes sense.